MessengerProj/src/main/java/com/b44t/messenger/LaunchActivity.java

```diff
@@ -753,7 +755,16 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
                 else {
                     switch (permissions[i]) {
                         case Manifest.permission.READ_CONTACTS:
-                            msg += "- " + ApplicationLoader.applicationContext.getString(R.string.PermissionContacts) + "\n\n";
+                            SharedPreferences preferences = ApplicationLoader.applicationContext.getSharedPreferences("general", Context.MODE_PRIVATE);
```

Comment:
instead of "general", we should use "mainconfig" which is also used to store most of the other preferences.